apps/tangle-dapp/containers/TransferTxContainer/TransferTxContainer.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+  AmountInput,
+  BridgeInputGroup,
+  Button,
+  Modal,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  RecipientInput,
+  Typography,
+} from '@webb-tools/webb-ui-components';
+import { TANGLE_DOCS_URL } from '@webb-tools/webb-ui-components/constants';
+import Link from 'next/link';
+import { FC, useCallback, useEffect, useState } from 'react';
+
+import { TANGLE_TOKEN_UNIT } from '../../constants';
+import useAccountBalances from '../../hooks/useAccountBalances';
+import useFormattedBalance from '../../hooks/useFormattedBalance';
+import useSubstrateTx, { TxStatus } from '../../hooks/useSubstrateTx';
+import convertToChainUnits from '../../utils/convertToChainUnits';
+import getTxStatusText from '../../utils/getTxStatusText';
+import { TransferTxContainerProps } from './types';
+
+const TransferTxContainer: FC<TransferTxContainerProps> = ({
+  isModalOpen,
+  setIsModalOpen,
+}) => {
+  const [amount, setAmount] = useState('');
+  const [recipientAddress, setRecipientAddress] = useState('');
+  const { transferrable: transferrableBalance } = useAccountBalances();
+
+  const formattedTransferrableBalance = useFormattedBalance(
+    transferrableBalance,
+    false
+  );
+
+  const {
+    execute: executeTransferTx,
+    status,
+    error: txError,
+  } = useSubstrateTx(
+    useCallback(
+      async (api) => {
+        const decimals = api.registry.chainDecimals[0];
+        const amountAsNumber = Number(amount);
+
+        // The amount is in the smallest unit of the token,
+        // so it needs to be converted to the appropriate amount
+        // of decimals.
+        const amountInChainUnits = convertToChainUnits(
+          amountAsNumber,
+          decimals
+        );
+
+        return api.tx.balances.transfer(recipientAddress, amountInChainUnits);
+      },
+      [amount, recipientAddress]
+    ),
+    true
+  );
+
+  // TODO: Likely would ideally want to control this from the parent component.
+  const reset = useCallback(() => {
+    setIsModalOpen(false);
+    setAmount('');
+    setRecipientAddress('');
+  }, [setIsModalOpen]);
+
+  // Reset state when the transaction is complete.
+  useEffect(() => {
+    if (status === TxStatus.Complete) {
+      reset();
+    }
+  }, [reset, status]);
+
+  const setMaxAmount = useCallback(() => {
+    if (formattedTransferrableBalance === null) {
+      return;
+    }
+
+    setAmount(formattedTransferrableBalance);
+  }, [formattedTransferrableBalance]);
+
+  const isReady = status !== TxStatus.Processing;
```

Comment:
Wrap these variables inside `useMemo` would be better since there are many state changes in this component